From 44a00b5b3f3229adb036b8339f5fa5ef98513f6d Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 19 Jun 2024 13:45:04 +0200
Subject: [PATCH 0920/1695] drivers: stm32_iac: handle stm32mp21 SoC
 particularities

Fix bindings file include and add stm32mp21x particularities for
iac_fbd_periph (No GPIOJ/GPIOK banks and ID changes).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8cf21e98ddc371b49fcce1a262c848a7f8f9989a drivers: stm32_iac: handle stm32mp21 SoC particularities]
Change-Id: I801a9d1628786937c1d50a6a8308822290349bee
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/388510
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/firewall/stm32_iac.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/core/drivers/firewall/stm32_iac.c b/core/drivers/firewall/stm32_iac.c
index 45c43e097..b341bac22 100644
--- a/core/drivers/firewall/stm32_iac.c
+++ b/core/drivers/firewall/stm32_iac.c
@@ -4,7 +4,7 @@
  */
 #include <drivers/stm32_iac.h>
 #include <drivers/stm32_rif.h>
-#include <dt-bindings/soc/stm32mp25-rifsc.h>
+#include <drivers/stm32mp_dt_bindings.h>
 #include <io.h>
 #include <kernel/boot.h>
 #include <kernel/dt.h>
@@ -124,6 +124,7 @@ static TEE_Result stm32_iac_parse_fdt(void)
 #define IAC_ILAC_ID(reg_val, offset)	(IAC_FIRST_ILAC_IN_REG(reg_val) + \
 					 IAC_EXCEPT_LSB_BIT(offset))
 
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
 static const uint32_t iac_fbd_periph[] = {
 	STM32MP25_RIFSC_PWR_ID, STM32MP25_RIFSC_GPIOA_ID,
 	STM32MP25_RIFSC_GPIOB_ID, STM32MP25_RIFSC_GPIOC_ID,
@@ -133,6 +134,17 @@ static const uint32_t iac_fbd_periph[] = {
 	STM32MP25_RIFSC_GPIOJ_ID, STM32MP25_RIFSC_GPIOK_ID,
 	STM32MP25_RIFSC_GPIOZ_ID
 };
+#endif
+#if defined(CFG_STM32MP21)
+static const uint32_t iac_fbd_periph[] = {
+	STM32MP21_RIFSC_PWR_ID, STM32MP21_RIFSC_GPIOA_ID,
+	STM32MP21_RIFSC_GPIOB_ID, STM32MP21_RIFSC_GPIOC_ID,
+	STM32MP21_RIFSC_GPIOD_ID, STM32MP21_RIFSC_GPIOE_ID,
+	STM32MP21_RIFSC_GPIOF_ID, STM32MP21_RIFSC_GPIOG_ID,
+	STM32MP21_RIFSC_GPIOH_ID, STM32MP21_RIFSC_GPIOI_ID,
+	STM32MP21_RIFSC_GPIOZ_ID
+};
+#endif
 
 static enum itr_return stm32_iac_itr(struct itr_handler *h __unused)
 {
-- 
2.47.2

